deps/0000-metrics-health-check.md

```diff
@@ -0,0 +1,175 @@
+container orchestrators (such as Kubernetes), and operational tools to interact with each component through standard HTTP 
+protocols. By consolidating metrics exposure and health check functionality into a single HTTP server per component, this 
+solution simplifies deployment, reduces infrastructure complexity.
+
+# Motivation
+
+Currently, the Dynamo runtime does not provide direct support for comprehensive metrics collection 
+or standardized health check mechanisms at the component level. While some metrics reporting to 
+Prometheus exists in the Rust frontend, there is no unified design for aggregating, querying, and managing metrics 
+across all Dynamo components. Additionally, there is no standardized way to check the health, liveness, and readiness of individual Dynamo components, making it difficult to monitor system health and implement proper load balancing and failover mechanisms.
+
+This lack of observability infrastructure creates several operational challenges:
+- No standardized health check mechanism for container orchestration systems (e.g., Kubernetes)
+- Limited visibility into component-level metrics and resource utilization
+- No centralized way to query and aggregate metrics/health across the distributed system
+
+## Goals
+
+* Implement a unified HTTP endpoint infrastructure for Dynamo components to expose metrics and health check endpoints
+* Enable customizable health checks through Python bindings while maintaining core health checks in Rust
+* Support standard observability patterns compatible with container orchestration systems
+
+
+## Requirements
+
+### REQ 1 Unified HTTP Endpoint Infrastructure
+
+The system **MUST** include a unified HTTP endpoint infrastructure for Dynamo components to expose metrics and health check endpoints
+
+### REQ 2 Performance Mrtics Requirements for worker nodes
+
+The metrics for worker nodes we want to monitor **MUST** include:
+- Inflight/Total Request: updated when a new request arrives (and finishes for inflight)
+- TTFT: reported at first chunk response
+- ITL: reported at each new chunk response
+- ISL: reported at first chunk response (TODO: report right after tokenization)
+- OSL: reported when requests finishes
+
+### REQ 3 Component Health Check Endpoints
+
+Each Dynamo component **MUST** expose HTTP endpoints for health monitoring:
+- `/health` - Overall component health status
+- `/liveness` - Component liveness probe
+- `/readiness` - Component readiness probe
+
+### REQ 4 Core Health Check Implementation
+
+The Rust runtime **MUST** implement basic health checks including:
+- etcd connectivity and lease status
+- NATS connectivity and service registration status
+
+### REQ 5 Extensible Health Check Framework
+
+The system **MUST** provide Python bindings that allow users to:
+- Register custom health check functions
+- Define component-specific health criteria
+- Customize health check response formats and thresholds
+
+
+
+# Architecture
```

Comment:
What I'd like to see is an overall diagram, let's make it Kubernetes-oriented, that shows a canonical graph (e.g. a disagg one) with the individual pods, and within them the components that are running, and the HTTP port/endpoints that are exposed.